pymeasure/instruments/hp/hp34401A.py

```diff
@@ -22,23 +22,36 @@
 # THE SOFTWARE.
 #
 
-import logging
+from warnings import warn
 from pymeasure.instruments import Instrument
 from pymeasure.instruments.validators import strict_discrete_set
 
-log = logging.getLogger(__name__)
-log.addHandler(logging.NullHandler())
-
 
 def _deprecation_warning(property_name):
     def func(x):
-        log.warning(f'Deprecated property name "{property_name}", use the "function_" '
-                    'and "reading" properties instead.')
+        warn(f'Deprecated property name "{property_name}", use the "function_" '
+             'and "reading" properties instead.', DeprecationWarning)
```

Comment:
See #672 on how to do proper deprecation:

It should be FutureWarning, not DeprecationWarnung, because users won't get a DeprecationWarning!

Additionally, please add a deprecated docstring (see #672) to the docstrings (you could write a single text and add it to the docstrings of all the deprecated properties.

Could you verify, that it works (run `make html` in the docs folder and open you local documentation)? For comparison, you might look at the [Adapter api documentation](https://pymeasure.readthedocs.io/en/latest/api/adapters.html#adapter-base-class).
Maybe the additional docstring text has to be a triple quoted one. Something like:
~~~python
deprecated_text = """
.. deprecated:: 0.12
    Use the "function_" and "reading" properties instead.
"""

The "deprecated since 0.11" is created from `.. deprecated:: 0.11`, which is mentioned in #672

Finally, please add a remark to the Changelog.